File: Factory/FactoryMethod/Java/src/PizzaStore/PizzaStore.java
package eu.svetlin.design.patterns.factory.factorymethod.PizzaStore;

import eu.svetlin.design.patterns.factory.factorymethod.Pizza.Pizza;

public abstract class PizzaStore {

    abstract Pizza createPizza(String type);
    public Pizza orderPizza(String type){
        Pizza pizza = createPizza(type);
        System.out.println("Making a " + pizza.getName());
        pizza.prepare();
        pizza.bake();
        pizza.cut();
        pizza.box();

        return pizza;
    }
}

File: Factory/FactoryMethod/TypeScript/src/PizzaStore/PizzaStore.ts
import Pizza from '../Pizza/Pizza';

abstract class PizzaStore {
  abstract createPizza(type: string): Pizza;

  orderPizza(type: string): Pizza {
    const pizza: Pizza = this.createPizza(type);
    console.log(`Making a ${pizza.getName()}`);
    pizza.prepare();
    pizza.bake();
    pizza.cut();
    pizza.box();

    return pizza;
  }
}

export default PizzaStore;
